helix-core/src/main/java/org/apache/helix/controller/rebalancer/strategy/CrushRebalanceStrategy.java

```diff
@@ -89,7 +89,16 @@ public ZNRecord computePartitionAssignment(final List<String> allNodes,
       long data = partitionName.hashCode();
 
       // apply the placement rules
-      List<Node> selected = select(topNode, data, _replicas, eventId);
+      List<Node> selected;
+      try {
+        selected = select(topNode, data, _replicas, eventId);
+      } catch (IllegalStateException e) {
+        String errorMessage = String
+            .format("Could not select enough number of nodes. %s partition %s, required %d",
+                _resourceName, partitionName, _replicas);
+        LogUtil.logError(Log, eventId, errorMessage);
```

Comment:
Seems the same in https://github.com/apache/helix/pull/346#discussion_r305574275

Reply:
Seems it is because there was a conflict in my two pull request which has not been approved yet. I will cancel this pull request and try to issue two separate non-conflicting pull requests.

Reply:
I fixed this issue in new PR.